test/unit/codegen/codegen_llvm_instance_struct.cpp

```diff
@@ -0,0 +1,172 @@
+// Utility to get initialized Struct Instance data
+//=============================================================================
+
+codegen::CodegenInstanceData generate_instance_data(const std::string& text,
+                                                    bool opt = false,
+                                                    bool use_single_precision = false,
+                                                    const size_t num_elements = 100) {
+    NmodlDriver driver;
+    const auto& ast = driver.parse_string(text);
+
+    // Generate full AST and solve the BREAKPOINT block to be able to generate the Instance Struct
+    AstVisitor().visit_program(*ast);
+    SymtabVisitor(true).visit_program(*ast);
+    NeuronSolveVisitor().visit_program(*ast);
+    SolveBlockVisitor().visit_program(*ast);
+    SymtabVisitor(true).visit_program(*ast);
+
+    codegen::CodegenLLVMVisitor llvm_visitor(/*mod_filename=*/"test",
+                                             /*output_dir=*/".",
+                                             opt,
+                                             use_single_precision);
+    llvm_visitor.visit_program(*ast);
+    llvm_visitor.print_module();
+    const auto& generated_instance_struct = llvm_visitor.get_instance_struct_ptr();
+    auto codegen_data = codegen::CodegenDataHelper(ast, generated_instance_struct);
+    auto instance_data = codegen_data.create_data(num_elements, 42);
+    return instance_data;
+}
+
+template <typename T>
+bool compare(T* instance_struct_data_ptr, const std::vector<T>& generated_data) {
+    std::vector<T> instance_struct_vector;
+    std::cout << "Generated data size: " << generated_data.size() << std::endl;
+    instance_struct_vector.assign(static_cast<T*>(instance_struct_data_ptr),
+                                  static_cast<T*>(instance_struct_data_ptr) +
+                                      generated_data.size());
+    for (auto value: instance_struct_vector) {
+        std::cout << value << std::endl;
+    }
+    return instance_struct_vector == generated_data;
+}
+
+//=============================================================================
+// Simple Instance Struct creation
+//=============================================================================
+
+SCENARIO("Instance Struct creation", "[visitor][llvm][instance_struct]") {
+    GIVEN("Instantiate simple Instance Struct") {
+        std::string nmodl_text = R"(
+            NEURON {
+                SUFFIX test
+                USEION na READ ena
+                RANGE minf, mtau
+                THREADSAFE : assigned GLOBALs will be per thread
+            }
+
+            STATE {
+                m
+            }
+
+            ASSIGNED {
+                v (mV)
+                celsius (degC)
+                ena (mV)
+                minf
+                mtau
+            }
+
+            BREAKPOINT {
+                SOLVE states METHOD cnexp
```

Comment:
@iomaganaris @pramodk I have noticed that when the NEURON does not have current, `SOLVE states METHOD cnexp` block is added to the kernel for loop:

```
for ( ... ) {
   SOLVE states METHOD cnexp
   ...
}
```

This does not affect code generation at all, but I am wondering if we should raise a ticket for this :)

Reply:
@georgemitenkov : I also so that somewhere and its a bug. Could you add ticket with some details?